Plugins/NativeEngine/Source/NativeEngine.cpp

```diff
@@ -1563,21 +1590,29 @@ namespace Babylon
         m_frameBufferManager.Unbind(frameBufferData);
     }
 
-    void NativeEngine::DrawIndexed(const Napi::CallbackInfo& /*info*/)
+    void NativeEngine::DrawIndexed(const Napi::CallbackInfo& info)
     {
-        //const auto fillMode = info[0].As<Napi::Number>().Int32Value();
+        const auto fillMode = info[0].As<Napi::Number>().Int32Value();
         //const auto elementStart = info[1].As<Napi::Number>().Int32Value();
         //const auto elementCount = info[2].As<Napi::Number>().Int32Value();
 
         // TODO: handle viewport
 
+        // TODO: support other fill modes
+        uint64_t fillModeState = 0; //indexed tri list
+        if (fillMode == 2)
+        {
+            fillModeState |= BGFX_STATE_PT_POINTS;
+        }
+
+
         for (const auto& it : m_currentProgram->Uniforms)
         {
             const ProgramData::UniformValue& value = it.second;
             bgfx::setUniform({it.first}, value.Data.data(), value.ElementLength);
         }
 
-        bgfx::setState(m_engineState);
+        bgfx::setState(m_engineState | fillModeState);
```

Comment:
bgfx::setstate merges 2 different concepts: rendering states (cullin, z test/write, color mask, ...) and draw primitive (lines, triangles, point,...). I want to keep those concepts sperate as much as I can. The rendering state is m_engineState but we have to set the primitive with SetState. So that call is performed just before submitting. It's not supposed to stick around. Next drawcall will call SetState.